packages/core/src/vm/vm.ts

```diff
@@ -50,151 +50,150 @@ interface VMConfig extends VMOptions {
 	providers?: Record<string, BlockProvider>
 }
 
-export class VM {
-	public static async initialize({ uri, spec, providers, ...options }: VMConfig): Promise<VM> {
-		const quickJS = await getQuickJS()
-		const runtime = quickJS.newRuntime()
-		const context = runtime.newContext()
-		runtime.setMemoryLimit(constants.RUNTIME_MEMORY_LIMIT)
+type SpecValidationResult = {
+	models: Record<string, Model>
+	actions: string[]
+	routes: Record<string, string[]>
+	contracts: Record<string, ethers.Contract>
+	contractMetadata: Record<string, ContractMetadata>
+	component: string | null
+	sources: Set<string>
```

Comment:
same with `actions`... I would rather have the whole thing be `parseExports(moduleHandle): { errors: string[]; warnings: string[]; exports: Exports | null }` (or even named `unwrapExports`?) where `Exports` is conceptually "exactly the things that are defined inside the spec, lifted up outside the VM" instead of a new type `SpecValidationResult` that has a mix of exports and derived metadata.

Could you try replacing `SpecValidationResult` with this?

```ts
interface Exports {
  models: Record<string, Model>
  actions: Record<string, QuickJSHandle>
  routes: Record<string, string>
  sources: Record<string, Record<string, QuickJSHandle>>
  contracts: Record<string, ContractMetadata>
  component: string | null
}
```

Reply:
Both of these comments sound great

Reply:
I'm thinking we should probably use names like `actionHandles` for this if it is a record containing handles, because we are using `actions` on the VM object as a publicly accessible property